misc/python/materialize/mzbuild.py

```diff
@@ -836,10 +837,30 @@ def ensure(self) -> None:
         """
         deps_to_build = [dep for dep in self if not dep.is_published_if_necessary()]
         self._prepare_batch(deps_to_build)
+
+        images_to_push = []
         for dep in deps_to_build:
             dep.build()
             if dep.publish:
-                spawn.runv(["docker", "push", dep.spec()])
+                images_to_push.append(dep.spec())
+
+        # Push all Docker images in parallel to minimize build time.
+        pushes: list[subprocess.Popen] = []
+        for image in images_to_push:
+            # Piping through `cat` disables terminal control codes, and so the
+            # interleaved progress output from multiple pushes is less hectic.
+            # We don't use `docker push --quiet`, as that disables progress
+            # output entirely.
+            push = subprocess.Popen(
+                f"docker push {shlex.quote(image)} | cat",
```

Comment:
Assuming our docker version is new enough `NO_COLOR=1` in environment should work too: https://github.com/moby/buildkit/pull/2954

Reply:
I suspect, though I did not check, that `NO_COLOR=1` disables escape codes related to coloring, but not to rendering the progress bars that update in place. That's a different set of escape codes.